src/java.base/share/native/libjava/jni_util.c

```diff
@@ -185,6 +185,16 @@ JNU_ThrowIOExceptionWithLastError(JNIEnv *env, const char *defaultDetail)
     JNU_ThrowByNameWithLastError(env, "java/io/IOException", defaultDetail);
 }
 
+/*
+ * Convenience method.
+ * Call JNU_ThrowByNameWithMessageAndLastError for java.io.IOException.
+ */
+JNIEXPORT void JNICALL
+JNU_ThrowIOExceptionWithMessageAndLastError(JNIEnv *env, const char *message)
+{
+    JNU_ThrowByNameWithMessageAndLastError(env, "java/io/IOException", message);
+}
```

Comment:
The additional function looks okay but I think the function description would be clearer if it were like this

```
/*
 * Throw java.io.IOException using a given message and the string
 * returned by getLastErrorString to construct the detail string.
 */
```

I see JNU_ThrowIOExceptionWithLastError also "Convenience method" but it's a function :-)